react-sdk/src/mcp/mcp-client.ts

```diff
@@ -135,8 +135,12 @@ export class MCPClient {
           : undefined;
 
       // Prevent re-entrant onclose during deliberate close by detaching
-      // the handler from the previous client instance.
-      const prevClient = this.client as any;
+      // the handler from the previous client instance without using `any`.
+      interface ClosableWithOptionalOnClose {
```

Comment:
@CharlieHelps why do you need to define a type here? isn't `prevClient` and `this.client` already typed to have `onclose`?